mm2src/lp_swap.rs

```diff
@@ -929,18 +932,25 @@ pub struct SavedTradeFee {
 
 impl From<SavedTradeFee> for TradeFee {
     fn from(orig: SavedTradeFee) -> Self {
+        // used to calculate locked amount so paid_from_trading_vol doesn't matter here
         TradeFee {
             coin: orig.coin,
             amount: orig.amount.into(),
+            paid_from_trading_vol: false,
```

Comment:
I'm afraid someone could initialize the `SavedTradeFee` fields directly.
What about adding
```rust
#[serde(default)]
paid_from_trading_vol: bool
```
to `SavedTradeFee` and use this `paid_from_trading_vol` flag in `locked_amount`?

Reply:
And also this flag is now ignored in the `TakerSwap::locked_amount` and `MakerSwap::locked_amount` methods if the swap has not been recovered from a file.

Reply:
Yeah, I agree, it will be more clear if we save this flag and then do not include the fee in `locked_amount` by checking it. Using the
```rust
let amount = if orig.paid_from_trading_vol {
            0.into()
        } else {
            orig.amount.into()
        };
        SavedTradeFee {
            coin: orig.coin,
            amount,
        }
```
looks quite magical :slightly_smiling_face: 
I'm going to change it now, thanks for the improvement proposal! :slightly_frowning_face:

Reply:
Done, could you recheck, please?